weave-js/src/components/PagePanelComponents/Home/Browse3/datasets/AddToDatasetDrawer.tsx

```diff
@@ -0,0 +1,309 @@
+      baseObjectClasses: ['Dataset'],
+    },
+    undefined,
+    true
+  );
+
+  useEffect(() => {
+    if (objectVersions.result) {
+      setDatasets(objectVersions.result);
+    }
+  }, [objectVersions.result]);
+
+  const handleNext = () => {
+    setCurrentStep(prev => Math.min(prev + 1, 2));
+  };
+
+  const handleBack = () => {
+    setCurrentStep(prev => Math.max(prev - 1, 1));
+  };
+
+  const projectId = `${entity}/${project}`;
+
+  const resetDrawerState = useCallback(() => {
+    setCurrentStep(1);
+    setSelectedDataset(null);
+    setFieldMappings([]);
+    setDatasetObject(null);
+    setError(null);
+    // Optionally reset the editContextRef if it supports a reset
+    // if (editContextRef.current && editContextRef.current.reset) {
+    //   editContextRef.current.reset();
+    // }
+  }, []);
+
+  const handleCreate = async () => {
+    if (!selectedDataset || !datasetObject) {
+      return;
+    }
+
+    setError(null);
+    try {
+      const {url, objectId} = await createDatasetVersion({
+        projectId,
+        selectedDataset,
+        datasetObject,
+        editContextRef,
+        tableUpdate,
+        objCreate,
+        router,
+        entity,
+        project,
+      });
+
+      toast(<DatasetPublishToast {...{url, objectId}} />, {
+        autoClose: 5000,
+        hideProgressBar: true,
+        closeOnClick: true,
+        pauseOnHover: true,
+      });
+
+      resetDrawerState();
+      onClose();
+    } catch (error) {
+      console.error('Failed to create dataset version:', error);
+      setError(
+        error instanceof Error ? error.message : 'An unexpected error occurred'
+      );
+    }
+  };
+
+  const isNextDisabled = currentStep === 1 && !selectedDataset;
+
+  const renderStepContent = () => {
+    switch (currentStep) {
+      case 1:
+        return (
+          <div>
+            <SelectDatasetStep
+              selectedDataset={selectedDataset}
+              setSelectedDataset={setSelectedDataset}
+              datasets={datasets}
+              selectedCallsCount={selectedCalls.length}
+            />
+            {selectedDataset && (
+              <SchemaMappingStep
+                selectedDataset={selectedDataset}
+                selectedCalls={selectedCalls}
+                entity={entity}
+                project={project}
+                fieldMappings={fieldMappings}
+                datasetObject={datasetObject}
+                onMappingChange={setFieldMappings}
+                onDatasetObjectLoaded={setDatasetObject}
+              />
+            )}
+          </div>
+        );
+      case 2:
+        return selectedDataset && datasetObject ? (
+          <EditAndConfirmStep
+            selectedCalls={selectedCalls}
+            fieldMappings={fieldMappings}
+            datasetObject={datasetObject}
+            editContextRef={editContextRef}
+          />
+        ) : null;
+      default:
+        return null;
+    }
+  };
+
+  return (
+    <ResizableDrawer
```

Comment:
It would be nice to be able to create a new dataset from this data. It feels odd that I have to first create a dataset in python before I can use this feature in the UI. Totally fine if that needs to be a followup